onnxoptimizer/passes/eliminate_slice_shape.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+#include "onnx/defs/tensor_util.h"
+#include "onnxoptimizer/pass.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+struct EliminateSliceShape final : public PredicateBasedPass {
```

Comment:
EliminateShapeSlice is better because the order of eliminated nodes in graph is Shape -> Slice

Reply:
what do you think of EliminateSliceAfterShape ? In this pass, Pattern `Shape -> Slice` be matched, we want to eliminate `Slice` op , but `Shape` will reserve.